packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/CanonicalViewProvider.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Generator.CSharp.Primitives;
+
+namespace Microsoft.Generator.CSharp.Providers
+{
+    public sealed class CanonicalViewProvider : TypeProvider
+    {
+        public CanonicalViewProvider(TypeProvider specView)
+        {
+            SpecView = specView;
+            SerializationViews = specView.SerializationProviders;
+        }
+
+        public TypeProvider SpecView { get; }
```

Comment:
can we not return `SpecView`?
once user have this in place, they can get everything from it